### PR TITLE
add Zest

### PR DESCRIPTION
**Pull request by VictorSanh**
Note: this is an example where all the splits don't have all the same features. The test split doesn't have the "domain" feature (always set to `null`). Jinja replaces it with `None`.
I still left the "domain" feature because it's secondary information and doesn't disrupt the flow.

### CODE REVIEW

**Comment by srush** [comment]
Do we allow different templates for splits? I thought it was just the sub-groups.

**Comment by VictorSanh** [comment]
> Do we allow different templates for splits? I thought it was just the sub-groups.

No, right now we don't have such a logic (different templates for different split).
For the sake of simplicity, I think we should check before applying the jinja whether all the fields are present. I can imagine that for many datasets, the test split just doesn't have the fields we use as labels.

File: templates/zest/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+      {{question}}|||
+
+      {% if answer[0] == "n/a" %}
+
+      Can''t answer
+
+      {% else %}
+
+      answer[0]
```
Comment (srush) [inline templates/zest/templates.yaml:72]: Needs to be in a {{ }}

File: templates/zest/templates.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+      {% endif %}'
+    name: gpt3instruct_format
+    reference: Template format from GPT3 instruct
+  846cc8ff-0527-4b2f-8da4-46613e915ff5: !Template
+    id: 846cc8ff-0527-4b2f-8da4-46613e915ff5
+    jinja: '{% if answer[0] != "n/a" %}{{context}}
+
+      Based on the previous passage, generate a question which has for answer {{answer[0]}}.
```
Comment (srush) [inline templates/zest/templates.yaml:81]: has "the following passage for an answer"

File: templates/zest/templates.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+      {{question}}|||
+
+      {% if answer[0] == "n/a" %}
+
+      Can''t answer
+
+      {% else %}
+
+      answer[0]
```
Comment (srush) [inline templates/zest/templates.yaml:119]: Needs to be {{ }}

File: templates/zest/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      |||
+
+      {{question}}
+
+      {% endif %}'
+    name: generate_the_question_with_domain
+    reference: Generate the question with domain
+  bdaf4f8a-2344-4e46-a52b-2045a080a4b2: !Template
+    id: bdaf4f8a-2344-4e46-a52b-2045a080a4b2
+    jinja: 'Answer this question about {{domain}} based on the context. If the question
```
Comment (srush) [inline templates/zest/templates.yaml:106]: domain is of the form "national_parks" I think you could replace the "_" with a space?

File: templates/zest/templates.yaml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+      What should be their answer based on this context: {{context}}|||
+
+      {% if answer[0] == "n/a" %}
+
+      I don''t know
+
+      {% else %}
+
+      answer[0]
```
Comment (srush) [inline templates/zest/templates.yaml:136]: Needs to be {{ }}

**Review by srush — CHANGES_REQUESTED** [review]

**Review by srush — APPROVED** [review]

**Pull request merged**